packages/core/test/features/st-structure.spec.ts

```diff
@@ -0,0 +1,548 @@
+import {
+    assertRule,
+    diagnosticBankReportToStrings,
+    shouldReportNoDiagnostics,
+    testStylableCore,
+} from '@stylable/core-test-kit';
+import {
+    CSSClass,
+    STStructure,
+    transformerDiagnostics,
+    STCustomState,
+} from '@stylable/core/dist/index-internal';
+import { expect } from 'chai';
+
+const transformerStringDiagnostics = diagnosticBankReportToStrings(transformerDiagnostics);
+const stStructureDiagnostics = diagnosticBankReportToStrings(STStructure.diagnostics);
+const stStateDiagnostics = diagnosticBankReportToStrings(STCustomState.diagnostics);
+const classDiagnostics = diagnosticBankReportToStrings(CSSClass.diagnostics);
+
+type FuncParameters<F> = F extends (...args: any[]) => any ? Parameters<F> : never;
+
+const spy = <T, N extends keyof T>(target: T, funcName: N) => {
+    const origin = target[funcName];
+
+    if (typeof origin !== 'function') {
+        throw new Error('spy only supports functions');
+    }
+
+    // type OriginType = FuncType<typeof origin>;
+    type OriginArgs = FuncParameters<typeof origin>;
+    const calls: OriginArgs[] = [];
+
+    // proxy
+    target[funcName] = ((...args: OriginArgs[]) => {
+        // record
+        calls.push([...args] as any);
+        // call original
+        return origin(...args);
+    }) as T[N];
+    return {
+        calls,
+        restoreSpy() {
+            target[funcName] = origin;
+        },
+    };
+};
+
+describe('@st structure', () => {
+    it('should warn experimental feature', () => {
```

Comment:
we don't want to warn we want to info

Reply:
this is a `console.warn`, it doesn't break any builds, and I prefer to to add `info` diagnostics to every call to the new API